src/sentry/tasks/relay.py

```diff
@@ -99,79 +97,146 @@ def update_config_cache(
         )
 
 
-def schedule_update_config_cache(
-    generate, project_id=None, organization_id=None, public_key=None, update_reason=None
-):
-    """
-    Schedule the `update_config_cache` with debouncing applied.
+@instrumented_task(
+    name="sentry.tasks.relay.build_config_cache",
+    queue="relay_config",
+    acks_late=True,
+    soft_time_limit=30,
+    time_limit=32,
+)
+def build_config_cache(public_key=None, trigger=None, **kwargs):
+    """Build a project config and put it in the Redis cache.
 
-    See documentation of `update_config_cache` for documentation of parameters.
+    This task is the primary way a new project config is computed.
+
+    :param organization_id: The organization for which to invalidate configs.
+    :param project_id: The project for which to invalidate configs.
+    :param generate: If `True`, caches will be eagerly regenerated, not only
+        invalidated.
+
+    Do not invoke this task directly, instead use :func:`schedule_build_config_cache`.
     """
+    validate_args(public_key=public_key)
 
-    if (
-        settings.SENTRY_RELAY_PROJECTCONFIG_CACHE
-        == "sentry.relay.projectconfig_cache.base.ProjectConfigCache"
-    ):
-        # This cache backend is a noop, don't bother creating a noop celery
-        # task.
-        metrics.incr(
-            "relay.projectconfig_cache.skipped",
-            tags={"reason": "noop_backend", "update_reason": update_reason},
-        )
-        return
+    sentry_sdk.set_tag("update_reason", trigger)
+
+    try:
+        keys = project_keys_to_update(public_key=public_key)
 
-    # validate_args(organization_id, project_id, public_key)
-    bools = sorted((bool(organization_id), bool(project_id), bool(public_key)))
-    if bools != [False, False, True]:
-        raise TypeError(
-            "One of organization_id, project_id, public_key has to be provided, not many."
+        compute_project_configs(keys)
```

Comment:
I'm afraid I changed this function.  And we'll also need to figure out how to make it disable projects instead of delete them.  maybe with an optional argument or so.